runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useProcessActionVariables.ts

```diff
@@ -0,0 +1,58 @@
+import { useMemo } from 'react';
+
+import { useModelerContext } from './useModelerContext';
+
+export interface ActionVariableObject {
+    name: string,
+    value?: string
+}
+
+interface ActionVariables {
+    inputActionVariables: ActionVariableObject[],
+    outputActionVariables: ActionVariableObject[]
+}
+
+const useProcessActionVariables = () => {
+    const context = useModelerContext();
+    const canvas = context?.modeler?.get('canvas');
+    const rootElement = canvas.getRootElement();
```

Comment:
what if canvas does not exist, maybe a guard return `if(!canvas) return []` or something like this. If there is not  canvas this can result in runtime error